subprojects/jvm-services/src/main/java/org/gradle/internal/jvm/inspection/DefaultJvmMetadataDetector.java

```diff
@@ -131,7 +132,10 @@ private static File javaExecutable(File jdkPath) {
     }
 
     private JvmInstallationMetadata parseExecOutput(File jdkPath, String probeResult) {
-        String[] split = probeResult.split(System.getProperty("line.separator"));
+        String[] split = Arrays.stream(probeResult.split(System.getProperty("line.separator")))
+                .filter(l -> l.startsWith(MetadataProbe.MARKER_PREFIX))
```

Comment:
:lipstick: nitpick so it reads the same way as the next statement
```suggestion
                .filter(line -> line.startsWith(MetadataProbe.MARKER_PREFIX))
```